streaming/base/dataset.py

```diff
@@ -765,6 +787,128 @@ def _resample_streams(self, epoch: int) -> Tuple[NDArray[np.int64], NDArray[np.i
         sample_ids = np.concatenate(sample_ids).astype(np.int64)
         return shuffle_units, sample_ids
 
+    def _generate_work_per_stream_batching(self, world: World, epoch: int,
+                                           sample_in_epoch: int) -> NDArray[np.int64]:
+        """Generate this epoch's arrangement of samples for ``per_stream`` batching.
```

Comment:
Can we also move this method to `partition` folder since this basically partition the samples across 5D (num physical nodes, ranks per node, workers per rank, batches per worker, batch size) ? this way we keep specific functionality in their respective modules.

Reply:
yeah sure!

Reply:
moving to separate `batching` folder instead.